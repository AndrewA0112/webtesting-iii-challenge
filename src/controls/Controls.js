import React from 'react';

const Controls = props => {
  const { locked, closed, toggleLocked, toggleClosed } = props;

  return (
    <div className="controls panel">
      <button disabled={!closed} onClick={toggleLocked} data-testid="lock" className="toggle-btn">
        {locked ? 'Unlock Gate' : 'Lock Gate'}
      </button>
      <button disabled={locked} onClick={toggleClosed} data-testid="open" className="toggle-btn">
        {closed ? 'Open Gate' : 'Close Gate'}
      </button>
    </div>
  );
};

export default Controls;
