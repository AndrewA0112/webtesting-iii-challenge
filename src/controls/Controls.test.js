import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react'

import Controls from './Controls'

describe('<Controls />', () => {
    it('Matches snapshot', () => {
        const tree = renderer.create(<Controls/>);
        expect(tree.toJSON()).toMatchSnapshot();
    })
})

describe('Closed Toggle', () => {
    it('the closed toggle button is disabled if the gate is locked', () => {
        const { getByTestId } = render(<Controls locked={true}/>)
        expect(getByTestId(/open/i).disabled).toBeTruthy();
    })
})

describe('Locked Toggle', () => {
    it('the locked toggle button is disabled if the gate is open', () => {
        const { getByTestId } = render(<Controls closed={false}/>)
        expect(getByTestId(/lock/i).disabled).toBeTruthy();
    })
})


