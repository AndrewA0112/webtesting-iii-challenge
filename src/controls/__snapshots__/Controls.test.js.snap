// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> Matches snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    data-testid="lock"
    disabled={true}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    data-testid="open"
  >
    Close Gate
  </button>
</div>
`;
